tests/variable_comparison_test.py

```diff
@@ -74,3 +75,109 @@ def test_identical():
 
     ds = sc.Dataset(data={'a': var})
     assert_export(sc.identical, ds['x', :], ds['x', :])
+
+
+@pytest.fixture
+def small():
+    return sc.scalar(1.0)
+
+
+@pytest.fixture
+def medium():
+    return sc.scalar(2.0)
+
+
+@pytest.fixture
+def large():
+    return sc.scalar(3.0)
+
+
+def test_eq(small, medium, large):
+    assert not (small == medium).value
+    assert not (small == medium.value).value
+    assert not (small.values == medium).value
+
+    assert (medium == medium).value
+    assert (medium == medium.value).value
+    assert (medium.value == medium).value
+
+    assert not (large == medium).value
+    assert not (large == medium.value).value
+    assert not (large.value == medium).value
+
+
+def test_ne(small, medium, large):
+    assert (small != medium).value
+    assert (small != medium.value).value
+    assert (small.values != medium).value
+
+    assert not (medium != medium).value
+    assert not (medium != medium.value).value
+    assert not (medium.value != medium).value
+
+    assert (large != medium).value
+    assert (large != medium.value).value
+    assert (large.value != medium).value
+
+
+def test_lt(small, medium, large):
+    assert (small < medium).value
+    assert (small < medium.value).value
+    assert (small.values < medium).value
+
+    assert not (medium < medium).value
+    assert not (medium < medium.value).value
+    assert not (medium.value < medium).value
+
+    assert not (large < medium).value
+    assert not (large < medium.value).value
+    assert not (large.value < medium).value
+
+
+def test_le(small, medium, large):
+    assert (small <= medium).value
+    assert (small <= medium.value).value
+    assert (small.values <= medium).value
+
+    assert (medium <= medium).value
+    assert (medium <= medium.value).value
+    assert (medium.value <= medium).value
+
+    assert not (large <= medium).value
+    assert not (large <= medium.value).value
+    assert not (large.value <= medium).value
+
+
+def test_gt(small, medium, large):
+    assert not (small > medium).value
+    assert not (small > medium.value).value
+    assert not (small.values > medium).value
+
+    assert not (medium > medium).value
+    assert not (medium > medium.value).value
+    assert not (medium.value > medium).value
+
+    assert (large > medium).value
+    assert (large > medium.value).value
+    assert (large.value > medium).value
+
+
+def test_ge(small, medium, large):
+    assert not (small >= medium).value
+    assert not (small >= medium.value).value
+    assert not (small.values >= medium).value
+
+    assert (medium >= medium).value
+    assert (medium >= medium.value).value
+    assert (medium.value >= medium).value
+
+    assert (large >= medium).value
+    assert (large >= medium.value).value
+    assert (large.value >= medium).value
+
+
+def test_comparison_with_str(medium):
+    assert (sc.scalar('a string') == sc.scalar('a string')).value
```

Comment:
How does this work, I cannot see where this is supported above?

Reply:
This is nothing new. I just wanted to add a test for this case as well, since there wasn't any. I guess the line you commented on does not really need to be tested in Python, I could remove it.
But the comparison with `str` directly makes sure that we don't convert the `str` to a `Variable`.

Reply:
Sorry, I misplaced a parenthesis when reading, I thought it compared `sc.scalar()` to a Python string.